Corona-Warn-App/src/main/java/de/rki/coronawarnapp/presencetracing/risk/CheckInMatching.kt

```diff
@@ -0,0 +1,75 @@
+package de.rki.coronawarnapp.presencetracing.risk
+
+import de.rki.coronawarnapp.eventregistration.checkins.CheckIn
+import de.rki.coronawarnapp.eventregistration.checkins.download.TraceTimeIntervalWarningPackage
+import de.rki.coronawarnapp.server.protocols.internal.pt.TraceWarning
+import de.rki.coronawarnapp.util.TimeAndDateExtensions.toLocalDate
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Deferred
+import kotlinx.coroutines.async
+import kotlinx.coroutines.withContext
+import org.joda.time.Duration
+import org.joda.time.Instant
+import kotlin.coroutines.CoroutineContext
+
+suspend fun launchMatching(
```

Comment:
I'd prefer a different name to prevent missunderstandings as this does not "launch" until `await` is called on it, devs unexperienced with coroutines may be caught off-guard.